packages/slice-machine/src/domain/customType.ts

```diff
@@ -129,6 +129,28 @@ export function getSectionSliceZoneConfig(
   return maybeSliceZone?.config ?? undefined;
 }
 
+// TODO add tests
+// We block user from deleteing section containing UID field,
+// so we can expect uid to be in the main section
+export function getUidField(customType: CustomType): UID | undefined {
+  const mainSectionEntry = getMainSectionEntry(customType);
+  const [_, mainSection] = mainSectionEntry ?? [];
+
+  return Object.values(mainSection ?? {}).find(
+    (field): field is UID => field.type === "UID",
+  );
+}
+
+// TODO add tests
+export function findSectionWithUidField(customType: CustomType): string | null {
+  const sectionEntries = getSectionEntries(customType);
+  const sectionWithUid = sectionEntries.find(([_, section]) =>
```

Comment:
❓#question

Can we have more than one section with UID? You are checking all sections but actually only get the first one anyway after with [0].